deep_qa/layers/attention/matrix_attention.py

```diff
@@ -1,10 +1,11 @@
 from copy import deepcopy
+from typing import Dict
 
 from keras import backend as K
 from overrides import overrides
 
 from ..masked_layer import MaskedLayer
-from ...common.params import Params
+from ...common.params import pop_choice_with_default
```

Comment:
Being nitpicky is ok - I'm nitpicky pretty often myself, as you have experienced...  If we can find something we're both happy with, it'll lead to a better API.  Here's a second try:

```python
def pop_choice(params, key, choices, default_to_first_choice: bool=False, other_default: Any=None):
  """
  Pops ``key`` from ``params``, ensuring that the popped value is in ``choices``.
  If ``key`` is not in ``params``, you can optionally default to the first item in
  ``choices``, or you can pass in a default value (which must also be in ``choices``).
  """
  if key not in params:
    if default is not None:
      if default not in choices:
        raise ConfigurationError
      return default
    elif allow_default:
      return choices[0]
    else:
      raise ConfigurationError("required key not specified; or some similar message")
  value = params.pop(key)
  if value not in choices:
    raise ConfigurationError()
  return value
```